app/src/main/java/me/guillaumin/android/osmtracker/db/DataHelper.java

```diff
@@ -65,6 +68,9 @@
 	 */
 	private Context context;
 
+	/** Statistics for all existing tracks */
+	private static TreeMap<Long, TrackStatistics> tracksStatistics = new TreeMap<Long, TrackStatistics> ();
```

Comment:
The fourth one is TrackManager, removeTrackStatistics in deleteTrack. I saw that track id's aren't reused but didn't want my code to rely on that.

Should it be a field of TracklistAdapter? If TracklistAdapter is to update its own statistics inside bind(...), wouldn't I have to write some additional complicated code to figure out which trackpoints aren't a part of the statistics yet? Also TracklistAdapter is recreated in TrackManager's onResume(), which means all statistics will be recalculated from scratch every time that thing is called, wouldn't it be a problem?

Also TrackDetail, you mean there should be two copies of the same data (and the same code), one in TracklistAdapter and another one in TrackDetail?